test/common/http/conn_manager_utility_test.cc

```diff
@@ -621,6 +621,183 @@ TEST_F(ConnectionManagerUtilityTest, UseXFFTrustedCIDRsEmptyXFFHeader) {
   EXPECT_EQ(headers.getForwardedForValue(), "10.3.2.1");
 }
 
+struct ExampleParameters {
+  bool use_remote_address{false};
+  uint32_t xff_num_trusted_hops{0};
+  std::string downstream_ip_address;
+  std::string received_xff_header;
+  std::string trusted_client_address;
+  std::string x_envoy_external_address;
+  std::string modified_xff_header;
+  std::vector<std::pair<std::string, unsigned char>> xff_trusted_cidrs;
+  bool envoy_internal{false};
+
+  ExampleParameters& useRemoteAddress(bool value) {
+    use_remote_address = value;
+    return *this;
+  }
+
+  ExampleParameters& xffNumTrustedHops(uint32_t value) {
+    xff_num_trusted_hops = value;
+    return *this;
+  }
+
+  ExampleParameters& downstreamIPAddress(const std::string& value) {
+    downstream_ip_address = value;
+    return *this;
+  }
+
+  ExampleParameters& receivedXFFHeader(const std::string& value) {
+    received_xff_header = value;
+    return *this;
+  }
+
+  ExampleParameters& trustedClientAddress(const std::string& value) {
+    trusted_client_address = value;
+    return *this;
+  }
+
+  ExampleParameters& xEnvoyExternalAddress(const std::string& value) {
+    x_envoy_external_address = value;
+    return *this;
+  }
+
+  ExampleParameters& modifiedXFFHeader(const std::string& value) {
+    modified_xff_header = value;
+    return *this;
+  }
+
+  ExampleParameters& envoyInternal(bool value) {
+    envoy_internal = value;
+    return *this;
+  }
+
+  ExampleParameters& addTrustedCidr(const std::string& cidr, unsigned char prefix_len) {
+    xff_trusted_cidrs.push_back(std::make_pair(cidr, prefix_len));
+    return *this;
+  }
+};
+
+class DocumentationExamples : public ConnectionManagerUtilityTest,
```

Comment:
Please add a comment describing where's the documentation that this should match.